Align server_theme breakpoints configuration with tailwind breakpoints

bboro opened the pull request:
This aligns the breakpoints config with tailwind breakpoints.

Updated readme to highlight important point on setup.

amitaibu commented:
To confirm, are we alreadu using the responsive image in the Article PEVB?

amitaibu commented:
I mean here - https://github.com/Gizra/drupal-starter/blob/2d16d8a1d060c4ad21ba23510748a8405f96b722/web/modules/custom/server_general/src/EntityViewBuilder/NodeViewBuilderAbstract.php#L76-L84 I think it should change?

bboro commented:
@amitaibu that looks like it passes the image's URL to the twig, so I highly doubt it's a responsive imageset with media queries etc. I don't think we're currently using any responsive image on drupal starter. This is just a bit of needed setup.

amitaibu commented:
@bboro In that case could you please wire it, so it's used.

bboro commented:
* Enabled Media and Responsive image
* Replaced Image reference `field_image` with Media reference `field_featured_image` in article node type
* Updated migration for articles, added media migration
* Updated hero frontend
* Added hero image styles and responsive image styles
* Wire up NodeArticle PEVB plugin with the new responsive image output
* Added `child-object-cover` and `child-object-contain` utility classes

| 640px (sm) | 768 (md) | 1024 (lg) | 1280 (xl) | 1920 |
| - | - | - | - | - |
| ![Screen Shot 2021-09-06 at 17 27 11](https://user-images.githubusercontent.com/24897663/132244818-2bd154e7-1e44-49d8-9827-21d5ebe22891.png) | ![Screen Shot 2021-09-06 at 17 27 17](https://user-images.githubusercontent.com/24897663/132244819-a8ba86f4-5e43-4e1f-8396-b252371cd6d3.png) | ![Screen Shot 2021-09-06 at 17 27 22](https://user-images.githubusercontent.com/24897663/132244820-42ea973d-e205-40af-8cf2-e79a2c16f629.png) | ![Screen Shot 2021-09-06 at 17 27 29](https://user-images.githubusercontent.com/24897663/132244826-769fc3a9-fa51-4d58-8187-786f32615901.png) | ![Screen Shot 2021-09-06 at 17 27 42](https://user-images.githubusercontent.com/24897663/132244828-2b1d04db-da32-4493-bc83-f7103c11f115.png) |

bboro commented:
Markup output of responsive image:

![image](https://user-images.githubusercontent.com/24897663/132245230-3fce89b6-eab6-4503-a450-d3db683b11a4.png)

amitaibu commented on README.md:
```diff
@@ -63,6 +63,21 @@ The directory structure:
 
+### Breakpoints and Responsive Images
+
+It is advised to use Drupal's Responsive image module.
```
Can you add a short paragraph about how you decide on the measurements (set the browser width, measure the img, etc')

bboro replied:
Added

amitaibu commented on README.md:
```diff
@@ -63,6 +63,21 @@ The directory structure:
+It is advised to use Drupal's Responsive image module.
+
+If there are new breakpoints added, or existing breakpoints updated in
+`server_theme/tailwind.config.js`, you must ensure to also update the drupal
```
```suggestion
`server_theme/tailwind.config.js`, you must ensure to also update the Drupal
```

amitaibu reviewed (approved):
Looks great, two small comments
